### PR TITLE
No longer make N calls to the volume API when listing instances + list bug fix.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Changed list instance code to load all volumes first, then find the volumes from this local list instead of calling the API each time in the loop.
- Removed code which loaded the status outside of a try / except, and resulted in the entire call list instance op failing if any status entries weren't found.

### CODE REVIEW

File: reddwarf/instance/models.py [hunk 1]
```diff
@@ -579,6 +579,18 @@ def find_server(instance_id, server_id):
     return find_server
 
 
+def create_volumes_list_matcher(volume_list):
+    # Returns a method which finds a volume from the given list.
+    def find_volumes(server_id):
```
Comment (ed-) [inline reddwarf/instance/models.py:584]: return lambda server_id: [volume for volume in volume_list if server_id in [attachment["server_id"] for attachment in volume.attachments]

I can't tell if the lambda is ugly or not

Comment (cp16net) [inline reddwarf/instance/models.py:584]: @ed- that is really ugly

**Comment by cp16net** [comment]
+1

**Comment by rnirmal** [comment]
+1

**Comment by TimSimpsonR** [comment]
I just realized I broke fake mode... one moment please...

**Comment by TimSimpsonR** [comment]
Fake mode fixed. Also, I added Ed's smoother version of the code with the list comp.

**Comment by imsplitbit** [comment]
+1

**Pull request merged**